File: scripts/import_to_supabase.js
/**
 * 使用方式：
 *   node scripts/import_to_supabase.js <SUPABASE_URL> <SUPABASE_KEY>
 *
 * 示例：
 *   node scripts/import_to_supabase.js https://xxx.supabase.co eyJhbGciOi...
 */

var https = require('https')
var path = require('path')

var url = process.argv[2]
var key = process.argv[3]

if(!url || !key){
  console.log('用法: node scripts/import_to_supabase.js <SUPABASE_URL> <SUPABASE_KEY>')
  console.log('示例: node scripts/import_to_supabase.js https://xxx.supabase.co eyJhbGciOi...')
  process.exit(1)
}

// 读取 products.js
var products
try{
  products = require(path.join(__dirname, '..', 'data', 'products'))
}catch(e){
  try{
    products = require(path.join(__dirname, '..', 'data', 'products.json'))
  }catch(e2){
    console.error('找不到 data/products.js')
    process.exit(1)
  }
}

if(!products || !products.length){
  console.error('产品数据为空')
  process.exit(1)
}

// 转换字段名（camelCase → snake_case）
var rows = products.map(function(p){
  return {
    id: p.id,
    category: p.category,
    brand: p.brand,
    name: p.name,
    color_name: p.colorName,
    hex: p.hex || '',
    price: Number(p.price) || 0,
    budget: p.budget || 'mid'
  }
})

// 分批发送（Supabase 单次限制 1000 条）
var batchSize = 100
var total = rows.length
var sent = 0

function sendBatch(){
  var batch = rows.slice(sent, sent + batchSize)
  if(!batch.length){
    console.log('\n✓ 全部导入完成，共 ' + total + ' 条')
    return
  }

  var body = JSON.stringify(batch)
  var apiUrl = url.replace(/\/+$/, '') + '/rest/v1/products'

  var req = https.request(apiUrl, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'apikey': key,
      'Authorization': 'Bearer ' + key,
      'Prefer': 'resolution=merge-duplicates'
    }
  }, function(res){
    var data = ''
    res.on('data', function(c){ data += c })
    res.on('end', function(){
      var ok = res.statusCode >= 200 && res.statusCode < 300
      if(ok){
        sent += batch.length
        process.stdout.write('\r  进度: ' + sent + '/' + total)
        sendBatch()
      } else {
        console.error('\n✗ 发送失败 (HTTP ' + res.statusCode + '):', data.slice(0, 200))
      }
    })
  })

  req.on('error', function(e){
    console.error('\n✗ 网络错误:', e.message)
  })

  req.write(body)
  req.end()
}

console.log('开始导入 ' + total + ' 条商品到 Supabase...')
sendBatch()

File: scripts/import_to_supabase.py
"""
使用方式: python3 scripts/import_to_supabase.py <SUPABASE_URL> <SUPABASE_KEY>
"""
import sys, json, urllib.request, os, re

url = sys.argv[1].rstrip('/')
key = sys.argv[2]

prods_path = os.path.join(os.path.dirname(__file__), '..', 'data', 'products.js')
with open(prods_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 去掉 module.exports =
content = content.replace('module.exports =', '').strip()
if content.endswith(';'):
    content = content[:-1]

# 去掉单行注释 //
content = re.sub(r'//.*', '', content)
# 去掉对象/数组末尾多余的逗号（JSON 不允许）
content = re.sub(r',\s*}', '}', content)
content = re.sub(r',\s*\]', ']', content)

# 去掉前后空格
content = content.strip()

products = json.loads(content)

if not products:
    print('产品数据为空')
    sys.exit(1)

# 转换字段名
rows = []
for p in products:
    rows.append({
        'id': p['id'],
        'category': p['category'],
        'brand': p['brand'],
        'name': p['name'],
        'color_name': p['colorName'],
        'hex': p.get('hex', ''),
        'price': float(p['price']),
        'budget': p.get('budget', 'mid')
    })

api_url = url + '/rest/v1/products'
headers = {
    'Content-Type': 'application/json',
    'apikey': key,
    'Authorization': 'Bearer ' + key,
    'Prefer': 'resolution=merge-duplicates'
}

total = len(rows)
sent = 0
batch_size = 100

print(f'开始导入 {total} 条商品到 Supabase...')

while sent < total:
    batch = rows[sent:sent + batch_size]
    data = json.dumps(batch).encode('utf-8')
    req = urllib.request.Request(api_url, data=data, headers=headers, method='POST')
    try:
        resp = urllib.request.urlopen(req)
        sent += len(batch)
        print(f'\r  进度: {sent}/{total}', end='', flush=True)
    except urllib.error.HTTPError as e:
        print(f'\n✗ 发送失败 (HTTP {e.code}): {e.read().decode()[:200]}')
        sys.exit(1)
    except urllib.error.URLError as e:
        print(f'\n✗ 网络错误: {e.reason}')
        sys.exit(1)

print(f'\n✓ 全部导入完成，共 {total} 条')
